debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -424,22 +389,63 @@ protected Object convertTimeWithZone(Column column, Field fieldDefn, Object data
         });
     }
 
-    protected Object convertTime(Column column, Field fieldDefn, Object data) {
-        if (adaptiveTimeMicrosecondsPrecisionMode) {
-            return convertTimeToMicrosPastMidnight(column, fieldDefn, data);
+    protected Object convertDate(Column column, Field fieldDefn, Object data) {
+        switch (temporalPrecisionMode) {
+            case ISOSTRING:
+                return convertDateToUtcIsoString(column, fieldDefn, data);
+            case ADAPTIVE_TIME_MICROSECONDS:
+            case ADAPTIVE:
+                return convertDateToEpochDays(column, fieldDefn, data);
+            default:
+                return convertDateToEpochDaysAsDate(column, fieldDefn, data);
         }
-        if (adaptiveTimePrecisionMode) {
-            if (getTimePrecision(column) <= 3) {
-                return convertTimeToMillisPastMidnight(column, fieldDefn, data);
-            }
-            if (getTimePrecision(column) <= 6) {
+    }
+
+    protected Object convertTime(Column column, Field fieldDefn, Object data) {
+        switch (temporalPrecisionMode) {
+            case ISOSTRING:
+                return convertTimeToUtcIsoString(column, fieldDefn, data);
+            case MICROSECONDS:
                 return convertTimeToMicrosPastMidnight(column, fieldDefn, data);
-            }
-            return convertTimeToNanosPastMidnight(column, fieldDefn, data);
+            case NANOSECONDS:
+                return convertTimeToNanosPastMidnight(column, fieldDefn, data);
+            case ADAPTIVE_TIME_MICROSECONDS:
+                // @TODO ? THIS IS INCONSISTEND WITH Timestamp
```

Comment:
This conforms to the documentation for both MySQL and PostgreSQL connectors so the TODO can be removed.

Reply:
removed